legacy/firmware/signing.c

```diff
@@ -1390,15 +1541,13 @@ static bool signing_check_orig_input(TxInputType *orig_input) {
     // (aka BIP-143 script code), which is what our code expects here in order
     // to properly compute the legacy transaction digest or BIP-143 transaction
     // digest.
-    if (!fill_input_script_sig(orig_input)) {
-      fsm_sendFailure(FailureType_Failure_ProcessError,
-                      _("Failed to derive public key."));
-      signing_abort();
+    if (!derive_node(orig_input) || !fill_input_script_sig(orig_input)) {
```

Comment:
couldn't we start using `script_pubkey` here?
(probably in a separate PR though)

or is there a possible conflict with host-provided script_pubkey?

Reply:
> couldn't we start using `script_pubkey` here? (probably in a separate PR though)

I think the problem is that the BIP-143 script code is not the actual scriptPubKey of this input, it's like the _legacy_ scriptPubKey that would correspond to the public key. I know, weird. See details of Item 5 in https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#specification.

> or is there a possible conflict with host-provided script_pubkey?

That wouldn't be a problem, because the host-provided script_pubkey is only for external inputs. For one we (logically) don't sign external inputs, for another we don't even support external inputs in T1.

Reply:
right, this thing!
it would have been much better if this was a separate global, but that's neither here nor there and might not be worth messing with anyway.

Reply:
It actually makes some sense to have this in script_sig, because when you compute the transaction digest for legacy signing, you put this scriptPubKey exactly in the spot where the scriptSig would belong in the serialized transaction.